### DISCUSSION TITLE
Where are model predictions saved by LightningCLI?

### DISCUSSION THREAD

**Discussion by cgoliver** [Lightning-AI/pytorch-lightning#14316] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I am trying to understand how the `predict` subcommand works with the LightningCLI.

As far as I can tell, it would be calling the lightning module's `predict_step` method. However, I don't know what happens after this. I see there is a `return_predictions` flag but not sure where or whether the output is stored somewhere. I am not finding it in the checkpoint directory.

Thanks in advance!

**Comment by tshu-w** [comment, marked as answer]
Hi, the return of subcommand is not saved in LightningCLI.
https://github.com/Lightning-AI/lightning/blob/0ca3b5aa1b16667cc2d006c3833f4953b5706e72/src/pytorch_lightning/cli.py#L623-L630

You can instance CLI with `run=False` or Hack the `LightningCLI` directly to make sure it return results.
```python
cli = LightningCLI(run=False)
cli.trainer.predict(...)
```

**Comment by mauvilsa** [comment]
For more context about this topic, read https://github.com/Lightning-AI/lightning/issues/12438
